src/ripple/rpc/handlers/LedgerHandler.cpp

```diff
@@ -48,35 +48,48 @@ Status LedgerHandler::check()
     if (auto s = lookupLedger (ledger_, context_, result_))
         return s;
 
-    bool bFull = params[jss::full].asBool();
-    bool bTransactions = params[jss::transactions].asBool();
-    bool bAccounts = params[jss::accounts].asBool();
-    bool bExpand = params[jss::expand].asBool();
-    bool bBinary = params[jss::binary].asBool();
+    bool const full = params[jss::full].asBool();
+    bool const transactions = params[jss::transactions].asBool();
+    bool const accounts = params[jss::accounts].asBool();
+    bool const expand = params[jss::expand].asBool();
+    bool const binary = params[jss::binary].asBool();
     bool const owner_funds = params[jss::owner_funds].asBool();
+    bool const queue = params[jss::queue].asBool();
 
-    options_ = (bFull ? LedgerFill::full : 0)
-            | (bExpand ? LedgerFill::expand : 0)
-            | (bTransactions ? LedgerFill::dumpTxrp : 0)
-            | (bAccounts ? LedgerFill::dumpState : 0)
-            | (bBinary ? LedgerFill::binary : 0)
-            | (owner_funds ? LedgerFill::ownerFunds : 0);
+    options_ = (full ? LedgerFill::full : 0)
+            | (expand ? LedgerFill::expand : 0)
+            | (transactions ? LedgerFill::dumpTxrp : 0)
+            | (accounts ? LedgerFill::dumpState : 0)
+            | (binary ? LedgerFill::binary : 0)
+            | (owner_funds ? LedgerFill::ownerFunds : 0)
+            | (queue ? LedgerFill::dumpQueue : 0);
 
-    if (bFull || bAccounts)
+    if (full || accounts)
     {
         // Until some sane way to get full ledgers has been implemented,
         // disallow retrieving all state nodes.
         if (! isUnlimited (context_.role))
             return rpcNO_PERMISSION;
 
         if (context_.app.getFeeTrack().isLoadedLocal() &&
-            !isUnlimited (context_.role))
+            ! isUnlimited (context_.role))
         {
             return rpcTOO_BUSY;
         }
-        context_.loadType = bBinary ? Resource::feeMediumBurdenRPC :
+        context_.loadType = binary ? Resource::feeMediumBurdenRPC :
             Resource::feeHighBurdenRPC;
     }
+    if (queue)
+    {
+        if (!ledger_ || !ledger_->open())
```

Comment:
Nit: space after exclamations. I only mention it because you corrected it above.